pkg/postgres/pgutils/retry.go

```diff
@@ -0,0 +1,37 @@
+package pgutils
+
+import (
+	"time"
+)
+
+const (
+	interval = 5 * time.Second
+	timeout  = 5 * time.Minute
+)
+
+// RetryQuery is used to specify how long to retry to successfully run a query
+// that fails with Transient errors
+func RetryQuery(fn func() error) error {
+	// Run query immediately
+	if err := fn(); !isTransientError(err) {
```

Comment:
I find it a bit subtle to rely on the fact that `isTransientError` returns `false` for `nil`. Maybe better to explicitly rewrite this as
```suggestion
	if err := fn(); err == nil || !isTransientError(err) {
```